dev-client/components/SiteMap.tsx

```diff
@@ -0,0 +1,47 @@
+import Mapbox, {Camera, MarkerView, UserLocation} from '@rnmapbox/maps';
+import React, {useEffect, useRef} from 'react';
+import {type DisplaySite} from '../datatypes/sites';
+import {type Position} from '@rnmapbox/maps/lib/typescript/types/Position';
+import {PermissionsAndroid} from 'react-native';
+import {IconButton, Icon} from 'native-base';
+// TODO: This vector icons module appears to be missing types
+import VIcon from 'react-native-vector-icons/MaterialIcons';
+
+type SiteMapProps = {
+  sites: DisplaySite[];
+  center: Position;
+};
+
+// REVIEW: This is a bad name, any suggestions?
+export default function SiteMap({sites, center}: SiteMapProps): JSX.Element {
```

Comment:
Anyone have any better ideas for the name here?

Reply:
somewhat broader answer, but i was thinking that we should create a lower-level reusable component which encodes a lot of the logic that will be shared between the different map views (e.g. how to handle the camera, setting user location, etc)

i think this would be helpful because the rnmapbox docs are pretty lacking, and if we could wrap all of our usage of rnmapbox in one component with fairly generic props, and then just call out to that component, we will limit the amount we need to read those docs, and also if the library stops being maintained there will be a clear API boundary between our code & the library

this generic component might have props like:
```ts
type TerrasoMapViewProps<T> = {
  ...,
  annotationData: T[],
  renderIcon: (_:T) => JSX.Element,
  ...
}
```

but we could do that in a future issue

Reply:
Totally agree! Personally my idea was to just start out with a clunky component and refactor it when I have a bit more experience using it, seeing what works and what doesn't. However if you have a clear idea with what you want, I think there is a separate issue to work on the reusable map component you could grab.